maintenance/migratestatus/migratestatus.go

```diff
@@ -0,0 +1,103 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"flag"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/google/go-github/github"
+	"k8s.io/test-infra/maintenance/migratestatus/migrator"
+)
+
+func main() {
+	var tokenFile, org, repo, copyContext, moveContext, retireContext, destContext string
+	var dryRun, continueOnError bool
+
+	flag.StringVar(&tokenFile, "tokenfile", "", "The file containing the token to use for authentication.")
+	flag.StringVar(&org, "org", "", "The organization that owns the repo.")
+	flag.StringVar(&repo, "repo", "", "The repo needing status migration.")
+	flag.BoolVar(&dryRun, "dry-run", true, "In dry run mode, no modifying actions will be taken.")
+	flag.BoolVar(&continueOnError, "continue-on-error", false, "Indicates that the migration should continue if context migration fails for an individual PR.")
+
+	flag.StringVar(&copyContext, "copy", "", "Indicates copy mode and specifies the context to copy.")
+	flag.StringVar(&moveContext, "move", "", "Indicates move mode and specifies the context to move.")
+	flag.StringVar(&retireContext, "retire", "", "Indicates retire mode and specifies the context to retire.")
+	flag.StringVar(&destContext, "dest", "", "The destination context to copy or move to. For retire mode this is the context that replaced the retired context.")
+	flag.Parse()
+
+	if org == "" {
+		errorfExit("'org' cannot be an empty string.\n")
+	}
+	if repo == "" {
+		errorfExit("'repo' cannot be an empty string.\n")
+	}
+	if tokenFile == "" {
+		errorfExit("'tokenfile' cannot be an empty string.\n")
+	}
+	tokenData, err := ioutil.ReadFile(tokenFile)
+	if err != nil {
+		errorfExit("Error loading token file: %v\n", err)
+	}
+
+	if destContext == "" {
+		errorfExit("'dest' is a required flag\n.")
+	}
+
+	var mode *migrator.Mode
+	modeMsg := "Exactly one mode must be specified [copy|move|retire].\n"
+	if copyContext != "" {
+		mode = migrator.CopyMode(copyContext, destContext)
+	}
+	if moveContext != "" {
+		if mode != nil {
+			errorfExit(modeMsg)
+		}
+		mode = migrator.MoveMode(moveContext, destContext)
+	}
+	if retireContext != "" {
+		if mode != nil {
+			errorfExit(modeMsg)
+		}
+		mode = migrator.RetireMode(retireContext, destContext)
+	}
+	if mode == nil {
+		errorfExit(modeMsg)
+	}
+
+	// Note that continueOnError is false by default so that errors can be addressed when they occur
+	// instead of blindly continueing to the next PR, possibly continuing to error.
+	m := migrator.New(*mode, strings.TrimSpace(string(tokenData)), org, repo, dryRun, continueOnError)
+
+	prOptions := &github.PullRequestListOptions{
+		State: "all",
+	}
+	if err := m.Migrate(prOptions); err != nil {
+		errorfExit("Error during status migration: %v\n", err)
+	}
+	glog.Flush()
+	os.Exit(0)
+}
+
+func errorfExit(format string, args ...interface{}) {
+	glog.Errorf(format, args...)
+	glog.Flush()
```

Comment:
Yeah, I was surprised it was needed, but without it the last logs weren't printing to stderr before exit.